src/api/plugins/tdbp/warren_tdbp/api.py

```diff
@@ -0,0 +1,169 @@
+"""Warren API v1 tdbp router."""
+
+import logging
+from datetime import datetime
+from typing import Optional
+
+from fastapi import APIRouter, status
+from ralph.exceptions import HTTPException, LrsClientException
+
+from .indicators import Cohort, Grades, Scores, SlidingWindow
+
+router = APIRouter(
+    prefix="/tdbp",
+)
+
+logger = logging.getLogger(__name__)
+
+
+@router.get("/window")
+async def get_sliding_window(
+    course_id: str,
```

Comment:
It seems that every endpoint is focused on a single `course_id`. 
What do you think about declaring `course_id` as a path parameter?
```suggestion
@router.get("/{course_id}/window")
async def get_sliding_window(
    course_id: str,
```
or
```suggestion
@router.get("/courses/{course_id}/window")
async def get_sliding_window(
    course_id: str,

Reply:
It works so we can leave it as is!